### ISSUE TITLE
How to update the temporal attention module?

### ISSUE THREAD

**Issue by xylshawn** [Dixin-Lab/BFVR-STC#2]
Thanks for sharing the codes. I'm curious about the update on the temporal attention parameters. During the forward phase,  the feature z_h is mapped to z_{h}^{TA} by the temporal attention module and pluses motion residuals to obtain the temporal latent. The temporal latent z_{h,t} is then quantized into z_{q,t}. The loss backward seems to not pass through the temporal attention.

**Comment by hhhh1138** [comment]
Thanks for your interest. 
Like other vector quantization methods, we use the **straight-through gradient estimator** to solve the non-differentiable problem of quantization operation. 

https://github.com/Dixin-Lab/BFVR-STC/blob/2c5b683d7b526b08bcb5cf4f07dfc7a9871012d2/basicsr/archs/vqgan3D_arch.py#L176

https://github.com/Dixin-Lab/BFVR-STC/blob/2c5b683d7b526b08bcb5cf4f07dfc7a9871012d2/basicsr/archs/vqgan3D_arch.py#L171-L173

The term `(z_q - z.detach()) ​** 2` allows gradients to flow through `z_q`, which depends on `z_t_q` and thus `z_t_flattened` (TA output). Gradients from both the commitment loss and KL loss can flow backward through its operations to update TA's parameters.
